### PR TITLE
bgpv1: Avoid ports from common ip_local_port_range in unit tests

### LINKED ISSUE #24691: CI: Travis CI - Pull Request (github.com/cilium/cilium/pkg/bgpv1/gobgp)
Check Travis CI - Pull Request is failing in the package github.com/cilium/cilium/pkg/bgpv1/gobgp

PR: https://github.com/cilium/cilium/pull/24688

Travis Job: https://app.travis-ci.com/github/cilium/cilium/builds/261984970

Log Output:

```
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃   FAIL  package: github.com/cilium/cilium/pkg/bgpv1/gobgp   ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
──────────────────────────────────────────────���─────────────────────────────────────────────────
                                                                                                
--- FAIL: TestPreflightReconciler (0.00s)
                                         
    --- FAIL: TestPreflightReconciler/local-port,_router-id_change (0.00s)
                                                                          
    reconcile_test.go:95: failed to create test BgpServer: failed starting BGP server: listen tcp4 0.0.0.0:45450: bind: address already in use
    --- FAIL: TestPreflightReconciler/local-port_change (0.00s)
                                                               
    reconcile_test.go:95: failed to create test BgpServer: failed starting BGP server: listen tcp4 0.0.0.0:45450: bind: address already in use
    --- FAIL: TestPreflightReconciler/no_change (0.00s)
                                                       
    reconcile_test.go:95: failed to create test BgpServer: failed starting BGP server: listen tcp4 0.0.0.0:45450: bind: address already in use
    --- FAIL: TestPreflightReconciler/router-id_change (0.00s)
                                                              
    reconcile_test.go:95: failed to create test BgpServer: failed starting BGP server: listen tcp4 0.0.0.0:45450: bind: address already in use
```

might be worth to retry or randomize port to prevent clashes with parallel tests which might randomly use this port?

### PR DESCRIPTION
**Pull request by rastislavs**
Unit tests should not use ports from the common default Linux `ip_local_port_range` (32768-60999), as those ports may be already used as a source port for an outgoing connection. This can cause intermittent failures in the CI manifesting as `bind: address already in use`.

E.g. in my Linux system:
```
$ cat /proc/sys/net/ipv4/ip_local_port_range 
32768	60999
```

This should avoid the CI issues for now, but the tests should be refactored - filed a follow-up issue in https://github.com/cilium/cilium/issues/26209

Fixes: https://github.com/cilium/cilium/issues/24691

### CODE REVIEW

File: pkg/bgpv1/manager/reconcile_test.go [hunk 1]
```diff
@@ -50,8 +50,8 @@ func TestPreflightReconciler(t *testing.T) {
 			name:           "no change",
 			routerID:       "192.168.0.1",
 			newRouterID:    "192.168.0.1",
-			localPort:      45450,
-			newLocalPort:   45450,
+			localPort:      30450,
```
Comment (rastislavs) [inline pkg/bgpv1/manager/reconcile_test.go]: Note that this is still not bullet-proof, as the `ip_local_port_range` is configurable. The other option would be to make the affected tests privileged and use common ports `< 1024`. Any preferences on this?

Comment (rastislavs) [reply pkg/bgpv1/manager/reconcile_test.go]: Filed a follow-up issue in https://github.com/cilium/cilium/issues/26209

**Review by harsimran-pabla — APPROVED** [review]
LGTM, thank you.

**Pull request merged**